src/onegov/org/models/search.py

```diff
@@ -159,5 +166,229 @@ def subset_count(self) -> int:
+            for vector in weighted_vectors[1:]:
+                combined_vector = combined_vector.op('||')(vector)
+        else:
+            combined_vector = func.to_tsvector(language, '')
+
+        return combined_vector
+
+    def generic_search(self) -> list['Searchable']:
+        doc_count = 0
+        results: list[Any] = []
+        language = locale_mapping(self.request.locale or 'de_CH')
+        ts_query = func.websearch_to_tsquery(language,
+                                             func.unaccent(self.web_search))
+        session = self.request.session
+
+        for base in self.request.app.session_manager.bases:
+            for model in searchable_sqlalchemy_models(base):
+                query = session.query(model)
+
+                if not self.request.is_logged_in:
+                    query = query.filter(
+                        model.fts_idx_data['es_public'].astext == 'True')
+
+                # as a member we only want to see public and member content
+                if self.request.is_member and hasattr(model, 'meta'):
+                    query = query.filter(
+                        model.meta['access'].astext.in_(('public', 'member')))
+
+                # as non-manager we only want to see public content
+                elif not self.request.is_manager:
+                    query = query.filter(
+                        model.fts_idx_data['es_public'].astext == 'True')
+
+                if session.query(query.exists()).scalar():
+                    weighted = (
+                        self._create_weighted_vector(model, language))
+                    rank_expression = func.coalesce(
+                        func.ts_rank(
+                            weighted,
+                            ts_query,
+                            0  # normalization, ignore document length
+                        ), 0).label('rank')
+                    query = (query.filter(model.fts_idx.op('@@')(ts_query))
+                             .add_columns(rank_expression))
+                    res = list(query.all())
+                    doc_count += len(res)
+                    results.extend(res)
+
+        # remove duplicates, sort by rank
+        results = list(set(results))
+        results.sort(key=lambda x: x[1], reverse=True)
+
+        self.nbr_of_docs = doc_count
+        self.nbr_of_results = len(results)
+
+        # remove rank column from results and return
+        return [r[0] for r in results]
+
+    def hashtag_search(self) -> list['Searchable']:
+        q = self.web_search.lstrip('#')
+
+        # Skip certain tables for hashtag search for better performance
+        results = [
+            doc for model in searchable_sqlalchemy_models(Base)
+            if model.es_type_name not in
+               ['attendees', 'files', 'people', 'tickets', 'users']
+            if model.es_public or self.request.is_logged_in  # type:ignore
+            for doc in self.request.session.query(model).all()
```

Comment:
```suggestion
            for doc in self.request.session.query(model)
```
You don't need to call `all` to iterate through a query, `all` will create a list for you, which you don't need here since you're iterating through the results.